test/IntegrationTests/AspNetTests.cs

```diff
@@ -65,5 +75,51 @@ public async Task SubmitsTraces()
 
         Assert.True(spans.Count >= 1, $"Expecting at least 1 span, only received {spans.Count}");
     }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    [Trait("Containers", "Windows")]
+    public async Task SubmitMetrics()
+    {
+        var collectorPort = TcpPortProvider.GetOpenPort();
+        var webPort = TcpPortProvider.GetOpenPort();
+        const int expectedMetricRequests = 1;
+
+        var testSettings = new TestSettings
+        {
+            MetricsSettings = new MetricsSettings { Port = collectorPort },
+        };
+
+        // Using "*" as host requires Administrator. This is needed to make the mock collector endpoint
+        // accessible to the Windows docker container where the test application is executed by binding
+        // the endpoint to all network interfaces. In order to do that it is necessary to open the port
+        // on the firewall.
+        using var fwPort = FirewallHelper.OpenWinPort(collectorPort, Output);
+        using var collector = new MockCollector(Output, collectorPort, host: "*");
+        using var container = await StartContainerAsync(testSettings, webPort);
+
+        var client = new HttpClient();
+
+        var response = await client.GetAsync($"http://localhost:{webPort}");
+        var content = await response.Content.ReadAsStringAsync();
+
+        Output.WriteLine("Sample response:");
+        Output.WriteLine(content);
+
+        var metricRequests = collector.WaitForMetrics(expectedMetricRequests, TimeSpan.FromSeconds(5));
+
+        using (new AssertionScope())
+        {
+            metricRequests.Count.Should().BeGreaterThanOrEqualTo(expectedMetricRequests);
+            var resourceMetrics = metricRequests.LastOrDefault().ResourceMetrics.Single();
```

Comment:
I think that you are correct and that the majority of our tests will be simple enough and fast enough that we only care about getting 1 metric request, and there will only be 1 metric request. I think that at least for now, if there are multiple metric requests, we may not care which request data that the test looks at especially if they are more of test of whether we successfully "enabled" an instrumentation library that we do not own.

The counter argument to this approach may come up when we start testing trace data via OTLP. The MockZipkinCollector has some helper methods that we take advantage of to ensure we are looking at the expected span data regardless of which request the span was included in. For traces we want to analyze all of the trace data that came in, instead of just looking at the latest request. We can either accept this break in symmetry between traces and metrics in the way we wait for and look at the data, or we try to keep them similar. I have not spent enough time trying out the approaches for testing traces via OTLP to get a feeling for what the approach should look like.

Reply:
That seems like an interesting idea to try out.